pkg/binlog/filename.go

```diff
@@ -30,14 +31,67 @@ var (
 	ErrInvalidBinlogFilename = errors.New("invalid binlog filename")
 )
 
-// VerifyBinlogFilename verify whether is a valid MySQL/MariaDB binlog filename.
-// valid format is `base + '.' + seq`.
-func VerifyBinlogFilename(filename string) error {
+// Filename represents a binlog filename.
+type Filename struct {
+	BaseName string
+	Seq      string
+}
+
+// LessThan checks whether this filename < other filename.
+func (f Filename) LessThan(other Filename) bool {
```

Comment:
binlog file with different base name can't  compare, how about add a warn log if base name is different

Reply:
currently, we treat _can't comparable_ as `false`, and verify the base in the caller (https://github.com/csuzhangxc/dm/blob/f07a3fb0ba81a84d07e93dd45261c144e6700116/pkg/streamer/file.go#L100)

Reply:
they are exported functions, it's better to return error if they can't comparable